django/contrib/auth/mixins.py

```diff
@@ -63,15 +63,52 @@ def handle_no_permission(self):
         )
 
 
-class LoginRequiredMixin(AccessMixin):
-    """Verify that the current user is authenticated."""
+class UserPassesTestMixin(AccessMixin):
+    """
+    Deny a request with a permission error if the test_func() method returns
+    False.
+    """
+
+    def test_func(self):
+        """
+        Write custom logic of your test and return either True or False
+        """
+        return True
```

Comment:
Why we change the logic here? This patch shouldn't change anything in the `UserPassesTestMixin` implementation.